persist/fs/read.go

```diff
@@ -108,28 +233,42 @@ func (r *reader) Open(namespace ts.ID, shard uint32, blockStart time.Time) error
 	if err := r.readCheckpointFile(shardDir, blockStart); err != nil {
 		return err
 	}
-	var infoFd, indexFd, dataFd, digestFd *os.File
+
+	var infoFd, digestFd *os.File
 	if err := openFiles(os.Open, map[string]**os.File{
 		filesetPathFromTime(shardDir, blockStart, infoFileSuffix):   &infoFd,
-		filesetPathFromTime(shardDir, blockStart, indexFileSuffix):  &indexFd,
-		filesetPathFromTime(shardDir, blockStart, dataFileSuffix):   &dataFd,
 		filesetPathFromTime(shardDir, blockStart, digestFileSuffix): &digestFd,
 	}); err != nil {
 		return err
 	}
 
 	r.infoFdWithDigest.Reset(infoFd)
-	r.indexFdWithDigest.Reset(indexFd)
-	r.dataFdWithDigest.Reset(dataFd)
 	r.digestFdWithDigestContents.Reset(digestFd)
 
 	defer func() {
 		// NB(r): We don't need to keep these FDs open as we use these up front
 		r.infoFdWithDigest.Close()
```

Comment:
Would it be better to not retain these fds in the struct?

Reply:
We don't actually retain the fds in the struct, we just retain the fd with digest wrappers so that we don't need to create them every time we open the file again (i.e. pooling the fd with digest wrappers).

Reply:
ah makes sense. would you mind putting a note down in the struct about the lifecycle of the wrappers